src/riak_test_escript.erl

```diff
@@ -236,8 +236,54 @@ parse_command_line_tests(ParsedArgs) ->
         end, [], lists:usort(DirTests ++ SpecificTests)).
 
 extract_test_names(Test, {CodePaths, TestNames}) ->
-    {[filename:dirname(Test) | CodePaths],
-     [filename:rootname(filename:basename(Test)) | TestNames]}.
+
+    RTTestPaths = get_test_paths(),
+    CommaSepTests = string:tokens(Test, [$,]),
+
+    WCTestsNoPrefix = [get_test_with_valid_prefix(TestA, RTTestPaths) || TestA <- CommaSepTests, get_test_type(TestA) =:= yes_wc_no_prefix],
```

Comment:
Looks like by building 4 lists you are touching each test 4 times.  How about move through the list once, but for each element in the perform different logic.  Maybe first check to see if it has a wildcard.  If so, then expand the list of test names.  You could use `lists:foldl` to build an accumulator of additional tests.  Before adding all of those tests to the list, then you could compare each new one to see if it has a prefix and see if it needs to be added to the code path and stripped from the test name.

Happy to discuss to clarify what I'm suggesting.